src/lcm/LifeCycleStates.cc

```diff
@@ -2496,3 +2681,504 @@ void LifeCycleManager::trigger_resize_failure(int vid)
         Quotas::quota_del(Quotas::VM, vm_uid, vm_gid, &deltas);
     });
 }
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void LifeCycleManager::trigger_disk_restore_success(int vid)
+{
+    trigger([this, vid]
+    {
+        Template vm_quotas_snp;
+        vector<Template *> ds_quotas_snp;
+        int uid = -1, gid = -1;
+
+        if ( auto vm = vmpool->get(vid) )
+        {
+            VirtualMachine::LcmState state = vm->get_lcm_state();
+
+            if (state == VirtualMachine::RESTORE)
+            {
+                uid = vm->get_uid();
+                gid = vm->get_gid();
+                vm->delete_snapshots(vm_quotas_snp);
+                vm->delete_non_persistent_disk_snapshots(vm_quotas_snp, ds_quotas_snp);
+
+                vm->set_state(VirtualMachine::POWEROFF);
+                vm->log("LCM", Log::INFO, "VM restore operation completed.");
+            }
+            else
+            {
+                vm->log("LCM", Log::ERROR, "VM restore success, VM in a wrong state");
+                return;
+            }
+
+            vmpool->update(vm.get());
+        }
+
+        if (!vm_quotas_snp.empty())
+        {
+            Quotas::quota_del(Quotas::VM, uid, gid, &vm_quotas_snp);
+        }
+
+        if ( !ds_quotas_snp.empty() )
+        {
+            Quotas::ds_del_recreate(uid, gid, ds_quotas_snp);
+        }
+
+    });
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void LifeCycleManager::trigger_disk_restore_failure(int vid)
+{
+    trigger([this, vid]
+    {
+        if ( auto vm = vmpool->get(vid) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that you are using unintended operations or assignments that could lead to logical errors, vulnerabilities, or undefined behavior. To prevent these issues, ensure that you follow best practices, such as using explicit comparisons and clear operator usage, and using parentheses to clarify intent. Learn more - https://cvefeed.io/cwe/detail/cwe-480-use-of-incorrect-operator